tests/SimpleSpec.hs

```diff
@@ -155,9 +166,39 @@ serverAddr = "127.0.0.1"
 testMsg :: ByteString
 testMsg = "This is a test message."
 
+unixAddr :: String
+unixAddr = "/tmp/network-test"
+
 ------------------------------------------------------------------------
 -- Test helpers
 
+-- | Establish a connection between client and server and then run
+-- 'clientAct' and 'serverAct', in different threads.  Both actions
+-- get passed a connected 'Socket', used for communicating between
+-- client and server.  'unixTest' makes sure that the 'Socket' is
+-- closed after the actions have run.
+unixTest :: (Socket -> IO a) -> ((Socket, SockAddr) -> IO b) -> IO ()
+unixTest clientAct serverAct = do
+    test clientSetup clientAct serverSetup server
+  where
+    clientSetup = do
+        sock <- socket AF_UNIX Stream defaultProtocol
+        connect sock (SockAddrUnix unixAddr)
+        return sock
+
+    serverSetup = do
+        sock <- socket AF_UNIX Stream defaultProtocol
+        bind sock (SockAddrUnix unixAddr)
```

Comment:
missing `unlink` before `bind`

Reply:
A more careful test might check whether the directory entry of that name (if it exists) is a socket (not a regular file or directory) and only unlink in that case, but "/tmp/network-socket" should be safe enough.  We should probably also unlink when the test is done.

Reply:
`unlink` is good. But is it available on Windows?

Reply:
It would be the same as the existing cleanup after the test.
```
E.tryJust (guard . isDoesNotExistError) $ removeFile unixAddr
```
Note also that Windows has no unix-domain sockets, so both the test and the unlink can be skipped on Windows.

Reply:
Done just in case.